S2VX.Game/Story/Note/GameNote.cs

```diff
@@ -104,22 +104,41 @@ public class GameNote : S2VXNote {
                 return true;
             }
 
-            base.UpdateNote();
+            UpdateColor();
+            UpdatePosition();
 
+            if (Time.Current >= HitTime + MissThreshold) {
+                RecordMiss();
+            }
+            return false;
+        }
+
+        protected override void UpdateColor() {
             var time = Time.Current;
+            var notes = Story.Notes;
+            // Fade in time to Show time
+            if (time < HitTime - notes.ShowTime) {
+                var startTime = HitTime - notes.ShowTime - notes.FadeInTime;
+                var endTime = HitTime - notes.ShowTime;
+                Alpha = Interpolation.ValueAt(time, 0.0f, 1.0f, startTime, endTime);
+            }
+            // Show time to Hit time with miss threshold time
+            // Hold the note at fully visible until after MissThreshold
+            else if (time < HitTime + MissThreshold) {
+                Alpha = 1;
+            }
+            // Hit time with miss threshold time to Fade out time
+            else if (time < HitTime + MissThreshold + notes.FadeOutTime) {
+                var startTime = HitTime + MissThreshold;
+                var endTime = HitTime + MissThreshold + notes.FadeOutTime;
+                Alpha = Interpolation.ValueAt(time, 1.0f, 0.0f, startTime, endTime);
+            } else {
+                Alpha = 0;
+            }
+
             if (time >= HitTime) {
-                // Hold the note at fully visible until after MissThreshold
-                var notes = Story.Notes;
-                var startFadeTime = HitTime + MissThreshold;
-                var endFadeTime = startFadeTime + notes.FadeOutTime;
-                var alpha = Interpolation.ValueAt(time, 1.0f, 0.0f, startFadeTime, endFadeTime);
-                Alpha = alpha;
                 Colour = Color4.Red;
```

Comment:
This will cause the note to remain Red if you start the hold late, is that fine?

Reply:
I think you meant to comment this in GameHoldNote.cs.

This issue is happening in master. In this checkin, I just replicated the behavior.

I don't think this behavior's fine though, as I'm confused if I missed it or think I'm like more punished for hitting notes late than early. I made an issue describing this issue: https://github.com/maxrchung/S2VX/issues/274. We can talk about solutions or visual changes in our next meeting.